templates/guides/rules.md

```diff
@@ -0,0 +1,169 @@
+
+Here's how a connection with a filter look like:
+
+```hcl
+resource "hookdeck_connection" "my_connection" {
+  source_id      = hookdeck_source.my_source.id
+  destination_id = hookdeck_destination.my_destination.id
+  rules = [
+    {
+      filter_rule = {
+        body = {
+          json = "{\"hello\":\"world\"}"
+        }
+        path = {
+          string = "/api/webhook"
+        }
+        query = {
+          boolean = false
+        }
+        headers = {
+          number = 10
+        }
+      }
+    }
+  ]
+}
+```
+
+As the Terraform provider expects a stringified JSON value for the JSON rule, there are some other approaches that you can use to configure your connection to your liking:
+
+- Using Terraform's [`jsonencoded`](https://developer.hashicorp.com/terraform/language/functions/jsonencode) for better readbility inline:
+
+```hcl
+resource "hookdeck_connection" "my_connection" {
+  source_id      = hookdeck_source.my_source.id
+  destination_id = hookdeck_destination.my_destination.id
+  rules = [
+    {
+      filter_rule = {
+        body = {
+          json = jsonencode({
+            hello = "world"
+          })
+        }
+      }
+    }
+  ]
+}
+```
+
+- Using Terraform's [`file`](https://developer.hashicorp.com/terraform/language/functions/file) to write your filter code in a separate file:
+
+```hcl
+resource "hookdeck_connection" "my_connection" {
+  source_id      = hookdeck_source.my_source.id
+  destination_id = hookdeck_destination.my_destination.id
+  rules = [
+    {
+      filter_rule = {
+        body = {
+          json = file("${path.module}/filters/my_connection_body.json")
+        }
+      }
+    }
+  ]
+}
+```
+
+## Transformation rule
+
+The transformation rule allows you to modify the payload of a webhook before it gets delivered to a destination.
+
+For more information on how to set up transformations, see our [transformation documentation](https://hookdeck.com/docs/transformations).
+
+To use transformation with Terraform, you need to create a new transformation before using it with your connection. Here's how that should look:
+
+```hcl
+resource "hookdeck_transformation" "my_transformation" {
+  name = "my_transformation"
+  code = "addHandler('transform', (request, context) => request);"
+}
+
+resource "hookdeck_connection" "my_connection" {
+  source_id      = hookdeck_source.my_source.id
+  destination_id = hookdeck_destination.my_destination.id
+  rules = [
+    {
+      transform_rule = {
+        transformation_id = hookdeck_transformation.my_transformation.id
+      }
+    }
+  ]
+}
+```
+
+As the transformation code also expects a stringified function handler, you can also keep your transformation code in a separate file.
+
+```hcl
+resource "hookdeck_transformation" "my_transformation" {
+  name = "my_transformation"
+  code = file("${path.module}/transformations/my_transformation.js")
+}
+```
+
```

Comment:
With `/transformations/my_transformation.js` as follows:

```js
addHandler('transform', (request, context) => request);
```

Reply:
ah yes, i missed adding a bit more instruction there for better clarity